Modify TokenV2 to add concurrent way of minting tokens using AggregatorV2

igor-aptos opened the pull request:
### Description

### Test Plan
<!-- Please provide us with clear details for verifying that your changes work. -->

msmouse committed:
P-txn executor

temp

JoshLind committed:
[Storage Service] Add type support for subscription requests.

JoshLind committed:
[Storage Service] Add subscription stream support.

JoshLind committed:
[Storage Service] Add subscription stream support.

JoshLind committed:
[Storage Service] Add tests for subscription streams.

igor-aptos committed:
Create aggregator_v2 branch with API in aggregator_v2.move

- add utility add()/sub() methods (#9370)
- Rename to max_value and make Aggregator template too (#9532)

JasperTimm committed:
Improve clarity of comments on token.move

igor-aptos committed:
Modify TokenV2 to add concurrent way of minting tokens using AggregatorV2

lightmark committed:
[event_v2] module event extension, attribute and extended check

lightmark committed:
[event_v2] module event extension, attribute and extended check

georgemitenkov committed:
[aggregator] New resolver and ID for aggregator  (#8941)

This PR adds the following:
    Aggregator ID can be either a table item or a u64 (unused);
    Aggregator does not use TableResolver anymore, and uses a custom one to enable value embedding;
    Aggregator test infrastructure has been moved around to allow for new IDs and enable cross-crate testing.

* [aggregator] New aggregator resolver

* [aggregator] Moving test infra around

* Addressed comments: Result<Option<u128>> --> Result<u128>

* Addressed comments

- ephemeral id is using u64 instead of u128
- history validation is factored out
- better error messages
- into_state_key --> as_state_key to operate on reference

* Require permission check before running run-indexer-tests

lightmark committed:
[eventv2] make v0/v1 to v1/v2

lightmark committed:
[eventv2] make v0/v1 to v1/v2

msmouse committed:
remove payer from StateValueMetadata

lightmark committed:
[event_v2] test

lightmark committed:
[event_v2] test

wrwg committed:
[events v2] Sample of how to check for event type attributes in the extended checker

wrwg committed:
[events v2] Sample of how to check for event type attributes in the extended checker

cryptomolot committed:
typo in functions.md

msmouse committed:
Track slot allocation fee on state metadata

msmouse committed:
enable state value metadata tracking for devnet

lightmark committed:
[eventv2] add module publishing validation

lightmark committed:
[eventv2] add module publishing validation

msmouse committed:
Track storage refund in the gas meter

msmouse committed:
add storage_fee_refunded to FeeStatement

Gonna expose the FeeStatement to outside of the VM for tests (and the
wider ecosystem for information)

JoshLind committed:
[Storage Service] Adopt subscription metadata and loop when serving
subscriptions.

lightmark committed:
[eventv2] add module event feature

lightmark committed:
[eventv2] add module event feature

cryptomolot committed:
Update package-upgrades.md

space added

cryptomolot committed:
Update unit-testing.md

typo

msmouse committed:
Explicitize that the txn_seq_num param on the epilogue is unused

With this we can go through all historical txns to make sure.

msmouse committed:
refund storage fee in epilogue

Commit:
[dag] Integrate dag fetcher with RB node handler

yemozer committed:
add guard field to struct

areshand committed:
temp

lightmark committed:
[eventv2] tests and fixes

lightmark committed:
[eventv2] tests and fixes

georgemitenkov committed:
[move][aggregator] Identifiers for aggregators & snapshots (#9075)

This PR allows exchanging values with identifiers in Move values.
Currently, the change happens at (de-)serialization time.

1. Resolvers have two methods now:
      - One to get bytes (the old flow)
      - One to get values with layouts (new flow)
    Both return Vec<u8> right now because we do not have
    Move values, but in general the second option can be
    used instead.

2. More tests.

3. Got rid of not necessary trait implementations and
    fixed some variable names.

4. ValueExchange is more generic now, and we can use a
    OnTagTransformation trait instead.

JoshLind committed:
[Dev Docs] Add note about indexer bootstrap support.

areshand committed:
[storage] update resource requriement

ibalajiarun committed:
[dag] Handle certified node in DAG Driver (#9312)

* [dag] Handle certified node in dag driver
* [dag] Trigger new round on handler start
* [dag] Use only the aptos_time_service crate

just-in-chang committed:
[NFT Metadata Crawler] Fix error for defaulting to original raw_image_uri (#9611)

* Fix error for defaulting to original raw_image_uri

* move db commit outside

xbtmatt committed:
[ts-sdk example] Adding an example for rotation offer capability and signer offer capability with signed structs (#9425)

* Adding the offer capability example. Uses signed structs in typescript.

* remove unnecessary aptosClient from getAccount call

Co-authored-by: Maayan <maayan@aptoslabs.com>

* Fixing network URL unnecessary code and potential unalignment between network/faucet url

* Shortening # hyphens in output string

* Moving the chainId to the bottom of the struct list since it's potentially undefined, and explaining in the sign struct function that the proof bytes must be in that specific order.

* Cleaning up the code and making it more readable

* Formatting

* Use a much cleaner and reusable serializable class instead of a struct

---------

Co-authored-by: Maayan <maayan@aptoslabs.com>

0xmaayan committed:
bump version to 1.18.0 (#9607)

wrwg committed:
[compiler-v2] Use livevar analysis to optimize file format generation (#9361)

* [compiler-v2] Use livevar analysis to optimize file format generation

This connects the live-variable analysis which is already present in the Move prover to the v2 bytecode pipeline. This information is then used in the file-format generator to make better decisions when to move or copy values.

Livevar is a standard compiler construction data flow analysis. It computes the set of variables which are alive (being used in subsequent reachable code) before and after each program point. The implementation from the prover uses our dataflow analysis framework and is [here](https://github.com/aptos-labs/aptos-core/blob/206f529c0c9d8488e27d2e50297178f0caf429a5/third_party/move/move-prover/bytecode/src/livevar_analysis.rs#L381). In this PR we create the first processing step in the bytecode pipeline with the new module `pipeline/livevar_analysis_step.rs` which forwards logical work to the existing analysis.

* Addressing reviewer comments.

georgemitenkov commented on aptos-move/framework/aptos-token-objects/sources/collection.move:
```diff
@@ -97,16 +102,50 @@ module aptos_token_objects::collection {
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// Fixed supply tracker, this is useful for ensuring that a limited number of tokens are minted.
+    /// and adding events and supply tracking to a collection.
+    struct ConcurrentFixedSupply has key {
```
ConcurrentFixedSupply and ConcurrentFixedSupply are the same, why we need both? We can keep one only, and the second `if` in increment can go away? If we want consistent errors having a boolean flag might be cleaner?

vusirikala replied:
So, we can have only one struct that tracks supply, and have an additional flag in the struct that says whether it is a fixed/unlimited supply?

georgemitenkov commented on aptos-move/framework/aptos-token-objects/sources/collection.move:
```diff
@@ -282,9 +331,35 @@ module aptos_token_objects::collection {
+                },
+            );
+            Union{ first: option::none(), second: option::some(aggregator_v2::snapshot(supply.total_minted)) }
+        } else if (exists<UnlimitedSupply>(collection_addr)) {
```
ConcurrentUnlimitedSupply

georgemitenkov commented on aptos-move/framework/aptos-token-objects/sources/token.move:
```diff
@@ -89,18 +95,23 @@ module aptos_token_objects::token {
         };
         move_to(&object_signer, token);
 
+        let token_appendix_1 = TokenAppendix1 {
```
Having Token::new can hide appendix creation?

cryptomolot committed:
typo spec-lang.md

typo

banool committed:
[TS SDK v2] Add Hex and HexInput types (#9595)

Co-authored-by: maayan <maayansavir@gmail.com>

igor-aptos committed:
making it such that we deprecate the old way, and have only the new way going forward

georgemitenkov committed:
[cleanup] Remove foreign contracts (#9623)

UIZorrot committed:
[Spec] Ensures of reconfiguration.move (#9383)

* init

* hp

* hp reconfig

* init

* fix comment

* fix comment

---------

Co-authored-by: chan-bing <zzywullr@gmail.com>

just-in-chang committed:
Revert "Revert "[NFT Metadata Crawler] Dockerize Parser (#9541)"" (#9622)

This reverts commit aa81b95e7c4bf7abc83e28b341ae6e8fdfdf7317.

alinush committed:
Clearer comments in `resource_account.move` (#9555)

vusirikala commented on aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move:
```diff
@@ -61,5 +61,9 @@ module aptos_framework::aggregator_v2 {
 
     public native fun snapshot<Element>(aggregator: &Aggregator<Element>): AggregatorSnapshot<Element>;
 
+    public native fun create_snapshot<Element: copy + drop>(value: Element): AggregatorSnapshot<Element>;
```
What's the potential use case of this? 
Do we want someone to be able to create a snapshot without an underlying aggregator? That would change the semantics and our underlying data structures as well.

ibalajiarun committed:
[dag] Integrate Order Rule with Dag Driver (#9452)

vusirikala commented on aptos-move/framework/aptos-framework/sources/object.move:
```diff
@@ -342,6 +342,13 @@ module aptos_framework::object {
+    public fun new_concurrent_event_handle<T: drop + store>(
+        object: &signer,
+    ): event::ConcurrentEventHandle<T> acquires ObjectCore {
+        event::new_concurrent_event_handle(create_guid(object))
```
As we are aiming for concurrency, should we use AUIDs instead of GUIDs?

vusirikala commented on aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move:
```diff
@@ -61,5 +61,9 @@ module aptos_framework::aggregator_v2 {
 
+    public native fun create_snapshot<Element: copy + drop>(value: Element): AggregatorSnapshot<Element>;
+
+    public native fun copy_snapshot<Element: copy + drop>(snapshot: &AggregatorSnapshot<Element>): AggregatorSnapshot<Element>;
```
Do we need this for Token V2? Are we copying snapshots anywhere, or are we only moving snapshots in Token v2?

wlawt committed:
[gas] convert output to InternalGas cost (#9603)

msmouse committed:
trivial: add FeatureFlag feature flags for deletion refund

vusirikala commented on aptos-move/framework/aptos-token-objects/sources/token.move:
```diff
@@ -89,18 +100,34 @@ module aptos_token_objects::token {
+        let token_appendix_1 = TokenAppendix1 {
+            index,
+        };
+        move_to(&object_signer, token_appendix_1);
```
So, we are creating token_appendix_1 independent of whether `concurrent_token_v2_enabled` feature flag is enabled?

vusirikala commented on aptos-move/framework/aptos-token-objects/sources/token.move:
```diff
@@ -89,18 +100,34 @@ module aptos_token_objects::token {
+
+        let deprecated_index = 0;
+        let index = if (features::concurrent_token_v2_enabled()) {
+            option::destroy_with_default(
```
I didn't completely understand what `destroy_with_default` is doing here. Are you incrementing total supply in the collection and destroying it? 
If concurrent_token_v2_enabled feature flag is enabled, then why are we assigning `index = 0` instead of `index = aggregator_v2::snapshot(total_supply)`?
If we are actually creating the index using `fn create_snapshot(u64)` instead of `fn snapshot(aggregator)`, what are we benefitting out of snapshots here?

JoshLind committed:
[GHA] Only run all perf tests on schedule.

vusirikala commented on aptos-move/framework/aptos-token-objects/sources/token.move:
```diff
@@ -254,6 +281,21 @@ module aptos_token_objects::token {
+    }
+
+    #[view]
+    public fun index_view<T: key>(token: Object<T>): u64 acquires Token, TokenAppendix1 {
```
Who calls `index_view` function? The indexer?

If we can combine `index_view` and `index_snapshot` methods, then we don't need to have the new native function `copy_snapshot` right? 
In fact, can we simply use the copy capability on AggregatorSnapshots instead of `copy_snapshot` native function?

wlawt committed:
[move-ir-compiler] add Nop token support (#9599)

ibalajiarun committed:
[dag] Integrate Dag Fetcher with Dag Driver (#9453)

* [dag] Integrate Dag Fetcher with Dag Driver
* [dag] Handle fetch response in dag handler

msmouse committed:
emit FeeStatement as a module event

banool committed:
[TS SDK v2] Add AccountAddress (#9564)

banool committed:
[CLI] Disallow "rotating" to same private key (#9546)

msmouse committed:
trivial: fix gas meter mutability

gelash committed:
Multi-versioned data structure specifically for aggregators / aggregator snapshots (#9472)

banool committed:
Remove unused deps for CLI (#9531)

davidiw committed:
Update developer-docs-site/docs/move/book/package-upgrades.md

xbtmatt committed:
[TS SDK example] Add a working example of converting a MultiEd25519 account to a MultiSig account (#9516)

* An example of converting a MultiEd25519 account to a MultiSig account

Cleaned up the file even more, added assertions and print statements at the end

Removing incorrect addresses

Formatting and clarifying the steps. Changing network to DEVNET from local

Removing confusing comment from prior version and adding comment about making sure secondsTilExpiration is set accordingly

Fixing file name

Fixing variable name

Making header comments less long

* Adding the ability to add metadata to the multisig account, since we were using empty arrays before. Asserted that the values on-chain match the input values at the end

* Using a much cleaner and reusable serializable class for the signed message

* Removing the unnecessary multisig as sender option

* Update e2e flow description at top of file

* Resolving rebase conflict and simplifying the signature construction

* Clarify the process of creating a MultiEd25519 account and that public keys are different from addresses

msmouse committed:
update compatibility test base

rustielin committed:
[dashboards] sync grafana dashboards

just-in-chang committed:
[NFT Metadata Crawler] Fix raw image uri parse error (#9628)

* Fix raw_image_uri parse bug

* upsert

davidiw committed:
Update developer-docs-site/docs/nodes/indexer-fullnode.md

wrwg committed:
[compiler v2] Fix bugs around compilation of vector code (#9636)

* [compiler v2] Fix bugs around compilation of vector code

This fixes a few bugs and closes #9629

- Usage of precompiled stdlib in transactional tests: v2 compiler currently (and probably never) supports precompiled modules, working around this
- Stack-based bytecode generator created wrong target on generic function calls
- Needed to make implicit conversion from `&mut` to `&` explicit in generated bytecode via Freeze operation

Added some additional tests while debugging this.

* Adding a new option `--print-bytecode` which can be provided to the `//# publish` and `//# run` transactional test runner command. This is the applied (for now) to the `sorter` test case only. Also introduced logic to map well-known vector functions to the associated builtin opcodes.

banool committed:
[TS SDK v2] Reject invalid SHORT form for special addresses in fromString (#9640)

* [TS SDK v2] Reject invalid SHORT form for special addresses in fromString

* Update account_address.ts

* Update account_address.ts

banool committed:
[TS SDK v2] Add equality methods for Hex and AccountAddress (#9641)

0xmigo committed:
Handle empty network address in CLI (#9576)

0xmigo committed:
[CLI][e2e] Tests create-resource-account and derive-resource-account-address (#8757)

* CLI e2e for resource account

* add account list to test resource account

aalok-t committed:
[prover] lock version dependency for z3 and boogie (#8718) (#9524)

Co-authored-by: Aalok Thakkar <aalok@Aaloks-MacBook-Pro.local>

msmouse committed:
storage refund tests

Commit:
Update workflows (#9650)

* Update semgrep.yaml to also run daily

* update semgrep rule

* fix workflows

* Update .github/workflows/semgrep.yaml

Co-authored-by: Balaji Arun <balaji@aptoslabs.com>

---------

Co-authored-by: Balaji Arun <balaji@aptoslabs.com>

yemozer committed:
add executor benchmark profiling

ibalajiarun committed:
[dag] bootstrap logic (#9455)

yemozer committed:
small fixes

yemozer committed:
lint fix

yemozer committed:
lint fix

yemozer committed:
lint fix

yemozer committed:
lint fix

yemozer committed:
lint fix

yemozer committed:
update cargo.lock

JoshLind committed:
[Storage Service] Add tests for stream looping.

UIZorrot committed:
[Spec] Ensures of resources.move (#9382)

* init

* fix md

* fix comment

msmouse committed:
remove storage_gas_units from FeeStatement

UIZorrot committed:
[Spec] Ensures of transaction_fee (#9460)

* hp trans_fee

* pre-pr

* add the condition that proposer == @vm_reserved

* fix comment

---------

Co-authored-by: chan-bing <zzywullr@gmail.com>

zekun000 committed:
[dag] support crash recovery for order rule

Add trait/impl for buffer manager adapter and implement the persist/recovery flow.

banool committed:
[TS SDK v2] Remove redundant code in AccountAddress.fromString (#9662)

areshand committed:
balanced replay-verify

gregnazario committed:
[docs] Rename Aptos Move CLI -> Aptos CLI (#9655)

brmataptos committed:
Thoroughly fix Issue 8875: Error out in compilation when encounter an unknown attribute (#9229)

Fix issue-8875 thoroughly: warning for unknown attributes, add --skip-attribute-check flag.  Omit warning on aptos_std library files, to avoid warning on some existing code.

A few tangential fixes:

* [third party/.../test, compiler] Fix Move.toml files to point to local files instead of network to avoid tests using old stdlib.  Oddly, this requires fixes to capitalization of std.  Fix the resolution warning to make it clear that std needs to be uncapitalized.

* [compiler] Hack to move-compiler to avoid warning about unknown attributes in aptos_std=0x1.
This avoids surprising warning on currently deployed library code.

* [compiler] test cases and stdlib source code updates for attributes checks PR

* [aptos stdlib] Now that we have verified that the move-compiler Aptos stdlib hack works, fix stdlib to use #[test_only] on test helper functions instead of #[testonly].  Leave struct attributes since it's a deployment problem to make it disappear.

areshand committed:
local debug

banool committed:
Make AccountAddress from_str conform to AIP-40, add from_str_strict (#9186)

* [TS SDK v2] Remove redundant code in AccountAddress.fromString

* Make AccountAddress from_str conform to AIP-40

* Use thiserror for AccountAddressParseError

manudhundi committed:
[Executor-benchmark] Make connected grps of txns fully conflicting grps (#9647)

[Executor-benchmark] Make connected grps of txns fully conflicting grps

Fully conflicting grps are more meaningful pattern for testing out the sharded
executor because they cannot be executed parallely (that is what we intend to benchmark).

Cmd:
cargo run -p aptos-executor-benchmark -- --block-size 100 --connected-tx-grps 5 --shuffle-connected-txns run-executor --main-signer-accounts 1000 --data-dir /tmp/some-db --checkpoint-dir /tmp/some-checkpoint --blocks 2

For a good random workload we would want 'num_accounts_per_grp' to be much greater than 'num_txns_per_grp'.
The command enforces 'num_signer_accounts' > '2 * num_txns_per_grp'.

lightmark committed:
[eventv2] address comments and forbid script emitting module events

lightmark committed:
[eventv2] address comments and forbid script emitting module events

just-in-chang committed:
[NFT Metadata Crawler] Stop parsing completely if token_uri has already been parsed (#9649)

* stop parsing completely if token_uri has already been parsed, add logs, add more panics (failed to write to gcs, failed to commit postgres tx)

* add comment to clarify

* error -> warn

* rename

junkil-park committed:
Fixed signer to be formatted properly (#8866)

* Fixed signer to be formatted properly

- fixed the issue in `string_utils`
- replaced `testonly` with `test_only`

* Add a feature flag

bchocho committed:
[Quorum Store] A couple metrics to observe backpressure better (#9239)

### Description

A couple metrics to observe backpressure better

grao1991 committed:
[Execution] Do not hold the Arc of committed block in execution. (#9674)

lightmark committed:
[eventv2] add script event verifier to forbid event emitting

lightmark committed:
[eventv2] add script event verifier to forbid event emitting

just-in-chang committed:
[NFT Metadata Crawler] Increase HTTP request for large files (#9677)

* retry time increase

* add to constants file

* remove unneeded comment

areshand committed:
Revert "swich to gcp"

This reverts commit 8f28f93c26fcb1c9ab1322fab4c00e0198ffbb67.

lightmark committed:
add #[event] to known attributes

lightmark committed:
add #[event] to known attributes

just-in-chang committed:
[NFT Metadata Crawler] Maintain image aspect ration on resize (#9688)

* maintain image aspect ration on resize

* fix lint

davidiw committed:
[python] Add some basic support for the CLI beginning with tests

UIZorrot committed:
[Spec] update spec for vesting.move (#9115)

* new vesting

* new vesting

* init

* fix comment

* fix md

* add comment

* add head

* add func total_accumulated_rewards time

* rm boogie

* fixed timeout

* close timeout function's verification

* rust lint fix

* del head

---------

Co-authored-by: chan-bing <zzywullr@gmail.com>

davidiw committed:
[python] Add support for building modules

With CLI support, we can now start adding better test coverage for
Python end-to-end tests.

msmouse committed:
trivial: add required feature to dependency to fix tests

`e2e-move-tests` when running alone doesn't pass because
`pub const GAS_UNIT_PRICE: u64 = 0;` is not in effect

davidiw committed:
[python] Add support for move unit testing from cli

lightmark committed:
[eventv2] api test

msmouse committed:
Revert "[eventv2] address comments and forbid script emitting module events"

This reverts commit e4a9c2cf41aeae5ae526f940f400909751bdc34f.

msmouse committed:
Revert "[eventv2] tests and fixes"

This reverts commit 4040b3d6b390ed75d344008f2dcd4b51c02bce13.

msmouse committed:
Revert "[eventv2] add module event feature"

This reverts commit 5a12afcb2e270217994ca5eafd309594f0a37dce.

msmouse committed:
Revert "[event_v2] test"

This reverts commit c93e61d61c9fd4884fcf0cd18ccc4f92cad9273a.

msmouse committed:
Revert "[eventv2] make v0/v1 to v1/v2"

This reverts commit 0ba554c725562a50952d0dc27b81149c74f97757.

msmouse committed:
Revert "[events v2] Sample of how to check for event type attributes in the extended checker"

This reverts commit 72cb7b4c7b9ed192962b3adfa6ad1459331afbf7.

msmouse committed:
Revert "[eventv2] add module publishing validation"

This reverts commit 7543f6e02fc0f2ecaf30c0830b4ce1d425ee4be5.

msmouse committed:
Revert "add #[event] to known attributes"

This reverts commit 5af30fb2adaa4247a711b2cbf75d81caa5412249.

msmouse committed:
Revert "[eventv2] add script event verifier to forbid event emitting"

This reverts commit 6a30bcdd429eca16668b599e674a493efd619cb7.

msmouse committed:
Revert "[event_v2] module event extension, attribute and extended check"

This reverts commit ed99bf02ec7e3578ad4669ea36b3515a0ff91ac6.

davidiw committed:
[scripts] fixes to arch linux (pacman)

* Archlinux / python / pip insist that you use the package manager
  solutions not python for pip
* The `libudev-dev` seems to exist in the default system install and under
  a wildly different name.

wrwg committed:
[compiler v2] Move the stackless bytecode crate out of prover (#9698)

This moves the `move-stackless-bytecode` crate out of its current location in the `move-prover` into the `move-model`. This is pure relocation. Subsequent PRs will split off prover specific code and move it back to the prover tree.

perryjrandall committed:
[framework] Reset the release yaml for upcoming 1.7 release

Test Plan: ran yq / generate proposals locally

```
target/performance/aptos-release-builder generate-proposals --release-config aptos-move/aptos-release-builder/data/release.yaml --output-dir output
```

wlawt committed:
[gas][docs] example guides (#9633)

georgemitenkov committed:
[refactoring] Relax trait bounds, reduce dependence on `StateView` (#9644)

API only needs move resolver, and not state view. Decoupling
`MoveResolverExt` into `AptosMoveResolver + StateView` so
that any future changes were not visible on API side.

1. Move API checks for resource groups where it should be.

2. Make trait bounds for config not require state view. Right now
state view is used all over the place although it seems this can
be avoided.

3. Reduce the usage of `MoveResolverExt`: e.g. session only
needs to talk to resolver.

4. State value metadata has its own resolver. This makes storage
accesses almost uniform (still need to fix `read_write_set`) which
should be implemented in storage adapter.

5. Some minor typo fixes and dead code removal.

ibalajiarun committed:
[dag] network sender implementation (#9456)

xosmig committed:
Add get_state_value_u128 method to TStateView (#9097)

UIZorrot committed:
[Spec] Ensures for voting (#9462)

* new_voting

* run lint

* little change

* fix md

* fix schema name

* fix error in simplemap

---------

Co-authored-by: chan-bing <zzywullr@gmail.com>

davidiw committed:
[python] wrap up large package publishing

This now supports chunking of even large modules. Will publish to
mainnet once approved.

added ability to test on a local testnet too

gelash committed:
[Block Executor] Refactor view & output traits & mvhashmap (#9659)

* [MVHashMap] convenient (shared) view state, aggregator v1 port

* [Block executor] new traits for processing the output

cryptomolot committed:
[typo*] aptos-token and wallets (#9661)

* Update aptos-token.md

typo "maxium"/maximum

* [typo*]Update wallets.md

dot added

wrwg committed:
[compiler v2] Stackless Bytecode Refactoring (#9713)

This is the 2nd step in the refactoring started in #9698.  This splits off prover specific parts from the crate `move-model/bytecode` into `move-prover/bytecode-pipeline`. Sharable dataflow analysis and transformation processors, like livevar and reaching definitions, stay. The tests have been split as well, and a common testing driver has been moved in its own test utility crate.

Github does not nicely show diffs like this, but this is functionally a no-op which only Moves code around.

msmouse committed:
integrate with executor-benchmark

danielxiangzl committed:
[DAG] Adding randomness field in dag node (#9687)

just-in-chang committed:
[NFT Metadata Crawler] Fix double slash URI parsing error, add ACKing PubSub message on receive (#9706)

* fix double slash uri parsing error, acking on message receive

* edit var names

* fix test

ngkuru committed:
[api] Add consistent API testing (#9577)

* [api] Add consistent API testing (#9146)

This commit introduces consistent API testing where we run a consistent load through the network mimicking various user flows. See proposal for more high level information.

Changes:

Created aptos-api-tester crate which runs 4 user flows:
Account creation
Coin transfer
NFT transfer
Module publishing, and outputs results to stdout.
Added token-client to the SDK with the methods needed to run Your First NFT.
mirroring the typescript SDK and the rust coin client by using existing builders
Added PartialEq to aptos_rest_client::Account for testing.
Added the crate to the tools docker image to run with GCP Cloud Run.

* [api] Add metrics to api tester (#9307)

This commit introduces logging tools on top of the consistent API testing introduced here. See proposal for more high level information.

Changes:

Added metrics pusher to aptos-api-tester with 4 histogram metrics for success, error, fail rates, and latency.
Added logs to tests instead of printing to stdout.

* [api] Add threads and timestamps to api tests (#9349)

This commit builds threads and run ID on top of the consistent API testing introduced here. See proposal for more high level information.

Changes:

Refactored tests into dedicated file tests.rs.
Switched to creating accounts for every tests instead of using the same and created pre test set up routines in testsetups.rs.
Added run ID start_time to metrics.

* [api] Add support for strictness level for consistency on API tests (#9444)

Description
This commit builds threads and run ID on top of the consistent API testing introduced here. See proposal for more high level information.

One problem we saw in the dashboard is that the problems we observed were regarding eventual consistency. We added support for adjusting the strictness level for such errors.

Changes:

Refactor of tests into tests module and decomposing into steps (setting up for next PR, individual step timing).
Persistent checks for information retrieval.
Added token support for Testnet faucet.

* [api] Add individual step timing to api tester (#9502)

This commit builds individual step timing on top of the consistent API testing introduced here. See proposal for more high level information.

This addition allows us to dive deeper on what the issue is should we detect any flow is consistently slower.

Changes:

Add timer macro in macros.rs.
Add helpers for emitting metrics and refactor process_result.
Put publish_module in its own thread by creating a tokio runtime.
Add sleeps between persistent checks.
Reduce API client sleep time to 100ms from 500ms.
Make test setups persistent.

* [api] Add view function testing to the API tester (#9658)

This commit adds a new test which tests a simple view function.

ibalajiarun committed:
[dag] e2e integration test (#9457)

michelle-aptos committed:
Update staking-pool-operations.md (#8685)

* Update staking-pool-operations.md

corrected CLI command to create staking_contract

previously the command created a direct pool type

* Update staking-pool-operations.md

add additional 8 0s

Commit:
Update Docker images (#9699)

Co-authored-by: gedigi <gedigi@users.noreply.github.com>

msmouse committed:
add metrics

lightmark committed:
[smart_vector] add destroy function for dropable T (#9434)

banool committed:
[CLI] Add support for setting faucet auth token (#9715)

* [CLI] Add support for setting faucet auth token

* Use FaucetClient

grao1991 committed:
Unify thread name, and combine all threads in the same thread pool in framegraph produced by cpu_profiler. (#9721)

gofenix committed:
Feat/pypi ci (#9512)

* add python sdk publish ci

* remove check version

msmouse committed:
add sorter

0xmaayan committed:
[TS SDK] Add account fetch queries (#9707)

* add account fetch queries

add account fetch queries

* rename getData to getInfo

* fix tests

---------

Co-authored-by: George Mitenkov <georgemitenk0v@gmail.com>

alinush committed:
Fix `Uniform` comments in `aptos-crypto/src/traits.rs` (#9733)

banool committed:
[Python SDK] Add support for faucet auth token (#9714)

* [Python SDK] Add support for faucet auth token

* Address feedback

larry-aptos committed:
add the header to data service. (#9651)

bchocho committed:
[Quorum Store] remove redundant checks (#9673)

### Description

remove handle_batch which does checks that ensure_max_limits already does

msmouse committed:
Track the refund on RespawnedSession

.. instead of on the meter. The benefit is when constructing the final
FeeStatement we are forced to contemplate on whether to issue or drop the
refund.

wrwg committed:
[compiler v2] Integration into package system and Aptos CLI (#9697)

This integrates the v2 compiler into the `move-package` code and makes it available to the `aptos` CLI. One can now compiler with the v2 compiler as in

```
aptos move compile --compiler-version=v2
```

However, compilation of already the Move stdlib fails because of #9330 (effecting bit operations in `fixed_point32.move`) and #9696 (effecting inline functions with lambdas in `vector.move`). Fixing those issues does not fit in this PR which is by itself not trivial.

Implementation wise, since the package system is build on top of internals of the v1 compiler, we provide a new v2 driver function parameter to the general entrypoint `compile_with_driver`. We also add a new flag for the compiler version which is plumbed the many layers of configurations from Aptos CLI down to Move compiler.

This PR also disables the use of the bytecode produced by v2 for production, by assigning a new experimental version number to it which is only allowed in testing mode.

This is tested manually.

zekun000 committed:
[consensus] replace direct send with reliable broadcast in buffer manager

movekevin committed:
[Framework] AIP-45: Allow soft burning objects

saharct committed:
Rename Token V2 to Digital Asset (DA) (#9730)

* Rename Token V2 to Digital Asset (DA)

* Update glossary

* Update additional_dict.txt

* Update developer-docs-site/docs/reference/glossary.md

Co-authored-by: Matt <90358481+xbtmatt@users.noreply.github.com>

* Update developer-docs-site/docs/reference/glossary.md

Co-authored-by: Matt <90358481+xbtmatt@users.noreply.github.com>

* Update digital-asset.md

* Update developer-docs-site/docs/standards/digital-asset.md

Co-authored-by: Matt <90358481+xbtmatt@users.noreply.github.com>

* Update fungible-asset.md

---------

Co-authored-by: Matt <90358481+xbtmatt@users.noreply.github.com>

msmouse committed:
adjustments

davidiw committed:
[python-sdk] add cli for publishing packages

in order to expedite availability, it seems prudent to have a wrapper
around our sdk code -- this exposes the publishing component... if this
becomes available in the Rust SDK, we can kill this functionality.

I also took advantage of this to further improve the package publishing
component of the SDK.

davidiw committed:
[python] add linting to examples

davidiw committed:
[python] update readme

msmouse committed:
more trace

msmouse committed:
hack fix

msmouse committed:
hack finalizer

msmouse committed:
avoid versioned_key as key

msmouse committed:
hashbrown

zjma committed:
A more efficient implementation of the multi-round block partitioner (#9488)

* init

* integrate into executor benchmark

* renames

* update

* more async drop

* update

* update

* update

* update

* update

* update

* save a vec of storage location

* update

* update

* update

* update

* update

* update

* update

* update

* update

* update

* update

* update

* exe bench partition timer

* parallelize add_edges__init

* update

* update

* prealloc r/wsets_by_txn_id

* update

* speculative bug fix

* add_edges bugfix

* give a large blockstm instance to the last shard

* par fix

* more v2 tests of equality to unsharded result

* add_edges another impl

* tweak

* determinism tests

* refactoring

* fmt

* tweak

* update

* v2 cmdline args for executor bench

* misc

* update tests

* clean up

* bugfix

* clean up

* more iterations in determinism tests

* clean up

* clean up

* comments

* API tweak

* clean up

* doc update

* clean up

* clean up

* rename

* polish partitionerv2 (#9619)

* save

* a quick bench...

* update

* update

* timer clean up

* all states in session

* update

* update

* update

* update

* parallel twds

* make_txn_with_dep as a func

* tweak

* tweak

* massive breakup

* tweak

* update

* rename

* update

* update

* tweaks

* tweaks

* timers back

* tweaks

* tweaks

* tweaks

* refactor

* renames

* update

* fmt

* doc

* tweak

* tweak

* comments

* comments

* tweak

* tweak

* tweak

* tweak

* tweak

* comment

* comment

* comment

* fmt

* renames

* fmt

* ut for start_txn_idxs

* renames

* comments

* comments

* fmt

* comments

* comments

* fmt

banool committed:
[indexer-docs] Add guide to parsing txns

davidiw committed:
[docs] remove legacy nft minting tool

* ambiguous adoption
* unsupported (last update was 8 months ago)
* not supporting the latest standard -- digital assets

wrwg committed:
[aptos-move] Make parallel BlockSTM comparison testing an opt-in behavior (#9726)

* [aptos-move] Make parallel BlockSTM comparision testing an opt-in behavior

Until now, each Move e2e test was executed two times: one time using sequential block executor, and then a second time using the parallel BlockSTM, comparing the result.

This is a large overhead of roughly 2/3 of the execution time of each test in our growing test suite. Here is a comparsion of running times for one e2e test:

```
With parallel:
cargo test --release test_fee_collection_and_distribution_for_burn_percentage  29.97s user 1.55s system 186% cpu 16.934 total

Without parallel:
cargo test --release test_fee_collection_and_distribution_for_burn_percentage  10.36s user 1.04s system 99% cpu 11.480 total
```

In this PR, we make the default behavior of parallel comparsion testing dependent on an environment variable E2E_PARALLEL_EXEC: only if this set will the old behavior be maintained. This allows to run those tests still locally or in a nightly run. Also functions are added to explicitly opt-in or opt-out of parallel execution. Possibly some of the existing e2e tests should be turned on again even as CI blocker for PE.

* Run CI Rust unit tests in release mode

banool committed:
[indexer-docs] Add basic guide to self hosting the Indexer API

movekevin committed:
Add Prover specs for new object burn/unburn functions

larry-aptos committed:
[monitoring] set the default latency if data absent. (#9748)

0xmigo committed:
[CLI] merge feature_branch/cli ledger_integration (#9703)

* [CLI] Add init from ledger account (#8300)

* add init from ledger account

Co-authored-by: Daniel Porteous (dport) <danielporteous1@gmail.com>

* refactor and added ledger account fetching from CLI

---------

Co-authored-by: Daniel Porteous (dport) <danielporteous1@gmail.com>

* [CLI][Ledger][easy] update ledger msg signing func (#8669)

* update ledger msg signing func

* fixes from comment

* [Rust SDK] Add HardwareWalletAccount for ledger integration (#8700)

* Add HardwareWalletAccount to SDK for signing

* [CLI] fix aptos ledger init (#8893)

* Refactor init with ledger

* Apply nit suggestions from code review

Co-authored-by: Daniel Porteous (dport) <daniel@dport.me>

---------

Co-authored-by: Daniel Porteous (dport) <daniel@dport.me>

* [CLI][1/n] Sign txn with ledger (#8745)

* sign txn with hardware wallet

* refactor submit transaction

* add 2 minutes exp time duration for signing transaction

* [CLI][Ledger][2/n] Add StatusCode to Aptos Ledger crate (#8898)

* Add Ledger status code for better error message

* fix nit

* [CLI][Ledger][3/n] Fix error handling (#8911)

* sign txn with hardware wallet

* fix error handling and some followup lint

* add serialize tag to derivation path (#8966)

* [CLI][Ledger] Add hardware wallet options (#9023)

* Add HardwareWalletOptions

* Add is_hardware_wallet fn to InitTool

* [CLI] add cli ledger doc (#9210)

* Add cli-ledger doc

* Apply suggestions from code review

Co-authored-by: Matt <90358481+xbtmatt@users.noreply.github.com>

* add blind signing explaination

---------

Co-authored-by: Greg Nazario <greg@gnazar.io>
Co-authored-by: Matt <90358481+xbtmatt@users.noreply.github.com>

* update derivation path format comment (#9465)

* fix changelog typo

* updated doc to inform user on limited ability of ledger nano s

* Update error printing

* correct validate_derivation_path

---------

Co-authored-by: Daniel Porteous (dport) <danielporteous1@gmail.com>
Co-authored-by: Daniel Porteous (dport) <daniel@dport.me>
Co-authored-by: Greg Nazario <greg@gnazar.io>
Co-authored-by: Matt <90358481+xbtmatt@users.noreply.github.com>

0xmigo committed:
bump CLI version to 2.1.0 (#9704)

* bump CLI version to 2.1.0

* add back unrelease section

* Update crates/aptos/CHANGELOG.md

Co-authored-by: Greg Nazario <greg@gnazar.io>

* update date to 8/22

* update date to 8/23

* update to 8/24

---------

Co-authored-by: Greg Nazario <greg@gnazar.io>

ibalajiarun committed:
[dag][bugfix] fix bitmask for incomplete first round (#9723)

bchocho committed:
[Quorum Store] Add back batches counter, and clean up around it (#9765)

### Description

Add back incrementing `RECEIVED_BATCH_COUNT` which was accidentally removed in #9673.
Clean up updating `RECEIVED_REMOTE_BATCH_COUNT` so it's done in the same place.

0xmaayan committed:
add support for fungible asset queries (#9747)

JoshLind committed:
[Dev Docs] Update node running instructions to mention testnet
requirement.

bchocho committed:
[Mempool] reduce mempool log noise (#9767)

### Description

Reduce the noise added by #9309. Make expected "errors" trace, and sample regardless.

larry-aptos committed:
[indexer grpc] sample the logs from data service (#9769)

junkil-park committed:
[Prover] Upgraded Boogie and adopted the new datatype syntax (#9180)

* Upgraded Boogie and adopted the new datatype syntax

* Added new-boogie-aptos-framework and new-boogie-aptos-stdlib

* Upgraded the boogie to 3.0.1

* Update aptos_token_sdk_builder.rs

sitalkedia committed:
[Execution][Benchmark] Add support for a pre-generating all transactions in the block-stm benchmark (#9740)

* Before partitioning

* Ready for review

* Fix ignoring no conflict txns

0xmaayan committed:
bump version to 1.19.0 (#9772)

grao1991 committed:
[Experimental] Add ThreadManager. (#9725)

banool committed:
Reorganize indexer docs to focus on the new indexer stack (#9763)

banool committed:
[indexer-docs] Add local development guide for the transcation stream service

banool committed:
[indexer-docs] Improve top level landing page

banool committed:
[Python SDK] Make from_str strict, add from_str_relaxed (#9642)

0xmigo committed:
[CLI] update cli script to have user use brew install if its macOS (#9648)

* update cli script to have user use brew install if its macOS

* update doc - remove macOS recommendation when installing from script

* Update workflows (#9650)

* Update semgrep.yaml to also run daily

* update semgrep rule

* fix workflows

* Update .github/workflows/semgrep.yaml

Co-authored-by: Balaji Arun <balaji@aptoslabs.com>

---------

Co-authored-by: Balaji Arun <balaji@aptoslabs.com>

---------

Co-authored-by: Gerardo Di Giacomo <gerardo@aptoslabs.com>
Co-authored-by: Balaji Arun <balaji@aptoslabs.com>

larry-aptos committed:
The processors are under aptos-indexer-processors now! (#9657)

Co-authored-by: Daniel Porteous <daniel@dport.me>

just-in-chang committed:
[NFT Metadata Crawler] Fix PubSub ACKing bugs and add logs (#9771)

* bug fix

* lint

* comments

msmouse committed:
enable module event v2 temporarily during framework update

msmouse committed:
limit max number of write ops

banool committed:
Move indexer docs under Build

cryptomolot committed:
[typo*]  (#9776)

* Update use-aptos-cli.md

* Update building-from-source.md

links do not work

* Update mint-nft-cli.md

typo

* Update mint-onchain-data.md

typo

* Update mint-onchain-data.md

typo

* Update 1-create-smart-contract.md

typo

* Update transaction-management.md

typo

* Update staking-pool-operations.md

typo

* Update staking-pool-operations.md

typo

* Update developer-docs-site/docs/guides/building-from-source.md

---------

Co-authored-by: David Wolinsky <isaac.wolinsky@gmail.com>

michelle-aptos committed:
Update staking-pool-operations.md (#9735)

* Update staking-pool-operations.md

add CLI command for updating commission

* Update developer-docs-site/docs/nodes/validator-node/operator/staking-pool-operations.md

---------

Co-authored-by: Greg Nazario <greg@gnazar.io>

banool committed:
Update redirects

JoshLind committed:
[Configs] Add seeds for testnet and mainnet.

brianolson committed:
Crate disentangle (#9608)

network/ crate -> network/framework/
(other network/** crates unaffected, no more intermingled crates)

brianolson committed:
config sanitize in test mode (#9664)

config sanitize in test mode

msmouse committed:
change how validate_upgrade deals with ReleaseEntry::FeatureFlag

Now it checks the diff only.

junkil-park committed:
[stake] allow updating commission percentage in the vesting module (#9719)

This commit allows updating the commission percentage for a vesting contract, which is the feature that the [AIP-48](https://github.com/aptos-foundation/AIPs/blob/main/aips/aip-48.md) proposes.

- Added the function `vesting::update_commission_percentage` and test cases

brianolson committed:
bump node.js to 18.x (#9799)

msmouse committed:
AnalyzedTransaction: fix that when transfering to self, write set has duplication

lightmark committed:
[smart_table] add clear and some nits (#9746)

lightmark committed:
[smart_table] a lot of inline functions added (#9752)

georgemitenkov committed:
[aggregator] New aggregator resolver (#9553)

- Aggregator does not use TableResolver anymore
- New aggregator APIs for V2
- Some basic test infrastructure changes to enable cross-crate testing.

Cherry-picks from #8941

msmouse committed:
disable module_event feature after framework update is done

zekun000 committed:
[dag] support crash recovery of dag driver

georgemitenkov committed:
Add parallel execution to aggregator tests (#9805)

msmouse committed:
Add test: limit is hit when trying to delete too many items

zjma committed:
Run transfer with hotspot in executor-benchmark (#9795)

* init

* fmt

* tweak

* tweaks

* more logs

sitalkedia committed:
[Execution][Storage] Split the ledger update to a separate phase (#9789)

georgemitenkov committed:
[format] Fix lint (#9809)

michelle-aptos committed:
Update staking.md (#9817)

update initialize-stake to create-staking-contract

msmouse committed:
add release step to enable AIP-32

rahxephon89 committed:
[Spec][move-prover] Multiple minor changes in specs and the prover lab (#9797)

* benchmark

* revise specs

Commit:
add ser/de to CompiledModule focused fuzzers (#9787)

* add repro script

* add ser/de to CompiledModule focused fuzzers

banool committed:
[indexer-docs] Update ratelimit, auth key information (#9737)

just-in-chang committed:
[NFT Metadata Crawler] Add metrics to parser (#9602)

github-actions[bot] committed:
[dashboards] sync grafana dashboards (#9660)

Co-authored-by: rustielin <rustielin@users.noreply.github.com>

vgao1996 committed:
[verifier] fix counting bug in signature checker v2 (#9754)

grao1991 committed:
Make sequence number atomic in LocalAccount, so some functions can take &self instead of &mut self. (#9822)

davidiw committed:
[features] fix flags (#9827)

* Some flags were not enabled by default even though they are marked for release
* One flag was off by 3
* Flags were in an inconsistent order making it hard to insure they were
  enabled..
* Update all goldens and fix a few broken tests

let's see if this blows up...

lightmark committed:
Update release.yaml to have aip44 (#9811)

sitalkedia committed:
Fix the block size in transaction benchmark (#9821)

msmouse committed:
impl Debug for AccessPath

gregnazario committed:
[node] Fix parsing of test seed (#9837)

sitalkedia committed:
Enable transaction shuffling in release.yaml for v1.7 release (#9819)

xbtmatt committed:
[TS-SDK] Fix the coin client transfer/checkBalance functions to allow custom CoinTypes (#9729)

* Updating coin client's transfer() and checkBalance() to check for a custom CoinType first before converting to a HexString address. Added the usage of a custom CoinType to the coin_client.test.ts unit test

* change `string | MaybeHexString` to MaybeHexString

Co-authored-by: Maayan <maayan@aptoslabs.com>

* Change `MaybeHexString` back to `string | MaybeHexString`. Adding comments to explain the usage of either a TypeTag CoinType or a fungible asset
address

* Make comment a little clearer

* Reduce comment line length to satisfy linter

---------

Co-authored-by: Maayan <maayan@aptoslabs.com>

lightmark committed:
rename function of event v2 (#9826)

vusirikala committed:
Aggregator snapshots native functions (#9766)

* Initial draft of aggregator snapshot native functions

* Create snapshot for string input

* read_snapshot and copy_snapshot for strings

* Add string_concat method

* Add unit tests

* Remove print statement

* Addressing PR comments

* Addresing PR comments

* Update release.yaml file

* Enable feature flag

* string_concat for u64 and u128

* Added a few unit tests

* Changing a few function names

* Minor changes

* Add e2e move tests

* rust lint

sasha8 committed:
Update CODEOWNERS (#9849)

davidiw committed:
[aptos-vm] prevent existing resources from becoming resource groups (#9829)

Left out a check that would allow users to accidentally migrate existing
resources into resource group members.
It likely also let them upgrade memberless structs into resource groups.

The worst case scenario is that data would be lost equivalently to
moving data to a burn address.

* using feature flag 30
* added tests for validating consistent behavior before and after
* updated the legacy resource group disabled test to actually validate
  before behavior...

msmouse committed:
fix bytecode-verifier-transactional-tests::tests

msmouse committed:
fix language_e2e_testsuite

msmouse committed:
fix move-unit-test

msmouse committed:
fix move-vm-transactional-tests

grao1991 committed:
Generate transactions in multiple threads. (#9842)

msmouse committed:
fix move-cli tests

sherry-x committed:
update release yaml (#9845)

banool committed:
[indexer-docs] Use absolute paths for links (#9838)

igor-aptos committed:
make name snapshot as well, and support index in name
